mongo/src/main/java/com/redhat/lightblue/mongo/crud/BasicDocSaver.java

```diff
@@ -288,59 +293,35 @@ private void updateDocs(CRUDOperationContext ctx,
                         try {
                             ctx.getFactory().getInterceptors().callInterceptors(InterceptPoint.PRE_CRUD_UPDATE_DOC, ctx, doc.inputDoc);
                             merge.merge(doc.oldDoc, doc.newDoc);
-                            Translator.populateDocHiddenFields(doc.newDoc, md);
+                            Translator.populateCaseInsensitiveHiddenFields(doc.newDoc, md);
                             updateAttemptList.add(doc);
                         } catch (Exception e) {
                             doc.inputDoc.addError(Error.get("update", MongoCrudConstants.ERR_TRANSLATION_ERROR, e));
                         }
                     } else {
                         doc.inputDoc.addError(Error.get("update",
-                                CrudConstants.ERR_NO_FIELD_UPDATE_ACCESS, paths.toString()));
+                                                        CrudConstants.ERR_NO_FIELD_UPDATE_ACCESS, paths.toString()));
                     }
                 }
                 LOGGER.debug("After checks and merge, updating {} docs", updateAttemptList.size());
                 if (!updateAttemptList.isEmpty()) {
-                    BulkWriteOperation bw = collection.initializeUnorderedBulkOperation();
+                    BulkUpdateCtx updateCtx=new BulkUpdateCtx();
+                    updateCtx.reset(collection);
```

Comment:
Does reset need to be public API? It seems like the contract is only, "call it immediately after you create it," since executing the update resets itself afterward. I think making it private and having the constructor take care of calling it would make it easier to use and less surprising since it wouldn't be constructed starting in an invalid state.